### ISSUE TITLE
feature request: more flexibility in atlas bounds/grid

### ISSUE THREAD

**Issue by danbjoseph** [fieldpapers/fieldpapers#179]
It would be helpful to be able to rotate the page orientation (so the top of the page is not north). For example if surveying buildings along both sides of a straight road that doesn't run N-S or E-W. 

It would also be a cool option to be able to choose the aspect ratio of the atlas grid and then have it fit the cells to the selected page size as best as possible.

**Comment by mojodna** [comment]
Rotation++

**Comment by mojodna** [comment]
(And more)

**Comment by ian-ross** [comment]
I've been thinking about this one a little bit, and was thinking about an alternative mode for defining an atlas (instead of a bounding box) where you draw a polyline on the map and Field Papers figures out an optimal way to make atlas pages to cover your polyline, rotated if necessary.  This might also work for GPS tracks too (#153).  In any case, I'll included rotation as a requirement for the new "server-independent" printer/scanner stuff I'm planning, which will at least make it possible to do this somehow.

**Comment by ian-ross** [comment]
We talked about this a little bit the other day, so it's definitely something that's on the radar.  Probably not for the first iteration of Lindsey's work, but it was in my mind as something we might do later if @linzjax makes good progress.

Just to be concrete about how this might work though, what is your use case here?  I had been thinking about "rotatable atlases" mostly in terms of surveying along roads, so I was thinking about implementing a completely different UI mode to the grid selection.  Here you would draw a polyline on the map and Field Papers would create atlas pages aligned along the polyline segments (with some overlap between line segments where you change direction, of course).  You'd be able to control the number of "rows" (pages across the polyline) and the size of an individual atlas page, then the number of "columns" (pages along the polyline) would be determined from that.  (This alternative line-drawing UI might also help with the request to support surveying along GPS tracks uploaded as GPX or KML.)

I got thinking about things in these terms because I couldn't think of how you would use a general capability to rotate an atlas, and the "survey along a road" case is what the issue was originally about.  Do you have a more general approach in mind though?

Incidentally, I had another related idea that might be quite easy to implement.  This was to add a checkbox to each individual atlas page in the page composer and allow the user to switch pages on or off individually.  This would be useful for cases where, for example, you make a big square map of a village that's next to a lake and half your atlas pages end up being blank because they're in the lake.  I don't know how often that might occur, but it struck me as potentially being useful in the context of this whole "mapping things that don't align nicely N-S/E-W" subject.

**Comment by mojodna** [comment]
Re: rotation, it's less about surveying along a road and more orienting "up" on the atlas to match an existing urban grid or a reference road that doesn't run east-west.

> Incidentally, I had another related idea that might be quite easy to implement. This was to add a checkbox to each individual atlas page in the page composer and allow the user to switch pages on or off individually. This would be useful for cases where, for example, you make a big square map of a village that's next to a lake and half your atlas pages end up being blank because they're in the lake. I don't know how often that might occur, but it struck me as potentially being useful in the context of this whole "mapping things that don't align nicely N-S/E-W" subject.

This would also be nice. The screencaps I've seen from @dalekunce definitely support this.

**Comment by ian-ross** [comment]
OK, that makes sense.  Thanks for the clarification.  We'll have a think about that...

**Comment by aawiseman** [comment]
Agreed on this, we experienced this when mapping a place that had a rough grid but the grid wasn't perfectly north-south.

**Comment by openbrian** [comment]
+1
